src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -77,8 +79,147 @@ public bool Equals(TagHelperAttribute other)
             return
                 other != null &&
                 string.Equals(Name, other.Name, StringComparison.OrdinalIgnoreCase) &&
-                Minimized == other.Minimized &&
-                (Minimized || Equals(Value, other.Value));
+                ValueStyle == other.ValueStyle &&
+                (ValueStyle == HtmlAttributeValueStyle.Minimized || Equals(Value, other.Value));
+        }
+
+        /// <inheritdoc />
+        public void WriteTo(TextWriter writer, HtmlEncoder encoder)
```

Comment:
We have very little unit test coverage of `WriteTo()`, `CopyTo()` and `MoveTo()` except in `TagHelperOutputTest`. That class does not check any value styles other than `DoubleQuotes`. Need more coverage, at least a few more cases in the `WriteTagHelper_InputData` data set.